Index: src\core\vm_actor.c
===================================================================
--- src\core\vm_actor.c
+++ src\core\vm_actor.c
@@ -75,16 +75,16 @@
         tx1 = SUBPX_TO_TILE(start_x + bounds->right);
         tx2 = SUBPX_TO_TILE(end_pos + bounds->right);
         if (tx2 < tx1) {
             tx2 = image_tile_width;
-        }            
+        }
     }
     while (ty1 != ty2) {
         if (tile_col_test_range_x(tile_mask, ty1, tx1, tx2)) {
             return (start_x > end_pos) ?
-                   TILE_TO_SUBPX(tile_hit_x) - bounds->left + TILE_TO_SUBPX(1) : 
+                   TILE_TO_SUBPX(tile_hit_x) - bounds->left + TILE_TO_SUBPX(1) :
                    TILE_TO_SUBPX(tile_hit_x) - EXCLUSIVE_OFFSET(bounds->right);
-        }                
+        }
         ty1++;
     }
     return end_pos;
 }
@@ -112,10 +112,10 @@
         }
     }
     while (tx1 != tx2) {
         if (tile_col_test_range_y(tile_mask, tx1, ty1, ty2)) {
-            return (start_y > end_pos) ? 
-                   TILE_TO_SUBPX(tile_hit_y) - bounds->top + TILE_TO_SUBPX(1) : 
+            return (start_y > end_pos) ?
+                   TILE_TO_SUBPX(tile_hit_y) - bounds->top + TILE_TO_SUBPX(1) :
                    TILE_TO_SUBPX(tile_hit_y) - EXCLUSIVE_OFFSET(bounds->bottom);
         }
         tx1++;
     }
@@ -210,9 +210,9 @@
             SET_FLAG(THIS->flags, MOVE_DIR_V);
         }
 
         THIS->PC -= (INSTRUCTION_SIZE + sizeof(idx));
-        return;        
+        return;
     }
 
     // Interrupt actor movement
     if (CHK_FLAG(actor->flags, ACTOR_FLAG_INTERRUPT)) {
@@ -269,9 +269,9 @@
         actor->pos.y += new_dir == DIR_UP ? -actor->move_speed : actor->move_speed;
 
         // Check for actor collision
         actor_t *hit_actor;
-        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) { 
+        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) {
             actor->pos.y = hit_actor->pos.y +
                 (new_dir == DIR_UP
                     ? hit_actor->bounds.bottom - actor->bounds.top + 1
                     : hit_actor->bounds.top - actor->bounds.bottom - 1
@@ -497,9 +497,11 @@
 
 void vm_actor_set_spritesheet(SCRIPT_CTX * THIS, INT16 idx, UBYTE spritesheet_bank, const spritesheet_t *spritesheet) OLDCALL BANKED {
     UBYTE * n_actor = VM_REF_TO_PTR(idx);
     actor_t * actor = actors + *n_actor;
-    load_sprite(actor->base_tile, spritesheet, spritesheet_bank);
+    actor->using_sprite_buffer = !actor->using_sprite_buffer;
+    UBYTE base_tile = actor->base_tile + (actor->using_sprite_buffer ? actor->reserve_tiles: 0);
+    load_sprite(base_tile, spritesheet, spritesheet_bank);
     actor->sprite.bank = spritesheet_bank;
     actor->sprite.ptr = (void *)spritesheet;
     load_animations(spritesheet, spritesheet_bank, ANIM_SET_DEFAULT, actor->animations);
     load_bounds(spritesheet, spritesheet_bank, &actor->bounds);
@@ -560,9 +562,11 @@
     gbs_farptr_t * params = VM_REF_TO_PTR(idxB);
     UBYTE spritesheet_bank = (UBYTE)(params->BANK);
     const spritesheet_t *spritesheet = params->DATA;
 
-    load_sprite(actor->base_tile, spritesheet, spritesheet_bank);
+    actor->using_sprite_buffer = !actor->using_sprite_buffer;
+    UBYTE base_tile = actor->base_tile + (actor->using_sprite_buffer ? actor->reserve_tiles: 0);
+    load_sprite(base_tile, spritesheet, spritesheet_bank);
     actor->sprite.bank = spritesheet_bank;
     actor->sprite.ptr = (void *)spritesheet;
     load_animations(spritesheet, spritesheet_bank, ANIM_SET_DEFAULT, actor->animations);
     load_bounds(spritesheet, spritesheet_bank, &actor->bounds);
@@ -645,9 +649,9 @@
     UBYTE test_actors = CHK_FLAG(attr, ACTOR_ATTR_CHECK_COLL_ACTORS);
 
     if (params->X == actor->pos.x) {
         // Already at destination
-        actor_set_anim_idle(actor);        
+        actor_set_anim_idle(actor);
         return;
     } else if (params->X < actor->pos.x) {
         // Moving left
         actor->pos.x -= actor->move_speed;
@@ -701,10 +705,10 @@
     THIS->waitable = 1;
 
     act_move_to_t * params = VM_REF_TO_PTR(idx);
     actor = actors + (UBYTE)(params->ID);
-    
 
+
     // Interrupt actor movement
     if (CHK_FLAG(actor->flags, ACTOR_FLAG_INTERRUPT)) {
         return;
     }
@@ -712,17 +716,17 @@
     UBYTE test_actors = CHK_FLAG(attr, ACTOR_ATTR_CHECK_COLL_ACTORS);
 
     if (params->Y == actor->pos.y) {
         // Already at destination
-        actor_set_anim_idle(actor);        
+        actor_set_anim_idle(actor);
         return;
     } else if (params->Y < actor->pos.y) {
-        // Moving upwards 
+        // Moving upwards
         actor->pos.y -= actor->move_speed;
 
         // Check for actor collision
         actor_t *hit_actor;
-        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) { 
+        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) {
             actor->pos.y += actor->move_speed;
             params->X = actor->pos.x;
             actor_set_anim_idle(actor);
             return;
@@ -739,9 +743,9 @@
         actor->pos.y += actor->move_speed;
 
         // Check for actor collision
         actor_t *hit_actor;
-        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) { 
+        if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) {
             actor->pos.y -= actor->move_speed;
             params->X = actor->pos.x;
             actor_set_anim_idle(actor);
             return;
@@ -819,14 +823,14 @@
     }
 
     if (!reached_y) {
         if (params->Y < actor->pos.y) {
-            // Moving upwards 
+            // Moving upwards
             actor->pos.y -= actor->move_speed;
 
             // Check for actor collision
             actor_t *hit_actor;
-            if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) { 
+            if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) {
                 actor->pos.y += actor->move_speed;
                 params->X = actor->pos.x;
                 reached_y = TRUE;
             }
@@ -841,9 +845,9 @@
             actor->pos.y += actor->move_speed;
 
             // Check for actor collision
             actor_t *hit_actor;
-            if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) { 
+            if (test_actors && (hit_actor = actor_overlapping_bb(&actor->bounds, &actor->pos, actor))) {
                 actor->pos.y -= actor->move_speed;
                 params->X = actor->pos.x;
                 reached_y = TRUE;
             }
@@ -870,9 +874,9 @@
     actor_t *actor;
 
     act_move_to_t * params = VM_REF_TO_PTR(idx);
     actor = actors + (UBYTE)(params->ID);
-    
+
     // Interrupt actor movement
     if (CHK_FLAG(actor->flags, ACTOR_FLAG_INTERRUPT)) {
         return;
     }
@@ -886,9 +890,9 @@
     actor_t *actor;
 
     act_move_to_t * params = VM_REF_TO_PTR(idx);
     actor = actors + (UBYTE)(params->ID);
-    
+
     // Interrupt actor movement
     if (CHK_FLAG(actor->flags, ACTOR_FLAG_INTERRUPT)) {
         return;
     }
@@ -902,7 +906,7 @@
     actor_t *actor;
 
     act_move_to_t * params = VM_REF_TO_PTR(idx);
     actor = actors + (UBYTE)(params->ID);
-    
+
     actor_set_anim(actor, actor->dir + N_DIRECTIONS);
 }
