Index: include\gbs_types.h
===================================================================
--- include\gbs_types.h
+++ include\gbs_types.h
@@ -1,5 +1,5 @@
-﻿#ifndef GBS_TYPES_H
+#ifndef GBS_TYPES_H
 #define GBS_TYPES_H
 
 #include <gbdk/platform.h>
 #include <gbdk/metasprites.h>
@@ -59,15 +59,15 @@
     uint16_t hscript_update, hscript_hit;
 
     // Collisions
     uint8_t collision_group;
-
+    uint8_t using_sprite_buffer;
     //Dynamic actor
     uint8_t actor_behavior_id;
     uint8_t actor_state;
-    int8_t actor_vel_x;
-    int8_t actor_vel_y;
-    struct actor_t *actor_parent;
+    int16_t actor_vel_x;
+    int16_t actor_vel_y;
+    uint8_t actor_linked_actor_idx;
 
 #ifdef DYNAMIC_ACTOR_ENABLE_MOVE_Z
     uint16_t pos_z;
     int8_t actor_vel_z;
@@ -211,5 +211,4 @@
 } palette_t;
 
 #endif
 
-
