Index: src\core\actor.c
===================================================================
--- src\core\actor.c
+++ src\core\actor.c
@@ -37,9 +37,9 @@
 
 const BYTE emote_offsets[] = {2, 1, 0, -1, -2, -3, -4, -5, -6, -5, -4, -3, -2, -1, 0};
 
 const metasprite_t emote_metasprite_8_16[]  = {
-    {8, 8, 0, 7}, {0, 8, 2, 7}, 
+    {8, 8, 0, 7}, {0, 8, 2, 7},
     {metasprite_end}
 };
 
 const metasprite_t emote_metasprite_8_8[]  = {
@@ -71,8 +71,9 @@
     player_moving           = FALSE;
     player_iframes          = 0;
     player_collision_actor  = NULL;
     emote_actor             = NULL;
+
     memset(actors, 0, sizeof(actors));
 }
 
 void player_init(void) BANKED {
@@ -84,9 +85,9 @@
 void actors_update(void) BANKED {
     actor_t *actor;
     static uint8_t screen_tile16_x, screen_tile16_y, screen_tile16_x_end, screen_tile16_y_end;
     static uint8_t actor_tile16_x, actor_tile16_y;
-    static uint8_t tmp_iterator; 
+    static uint8_t tmp_iterator;
     static FASTUBYTE actor_flags;
 
     // Convert scroll pos to 16px tile coordinates
     // allowing full range of scene to be represented in 7 bits
@@ -146,9 +147,9 @@
                         SET_FLAG(actor->flags, ACTOR_FLAG_DISABLED);
                     } else {
                         if (CHK_FLAG(actor_flags, ACTOR_FLAG_DISABLED)) {
                             CLR_FLAG(actor->flags, ACTOR_FLAG_DISABLED);
-                        }                        
+                        }
                         deactivate_actor_impl(actor);
                     }
                 } else {
                     SET_FLAG(actor->flags, ACTOR_FLAG_DISABLED);
@@ -202,9 +203,9 @@
     for (actor = actors_active_tail; (actor); actor = actor->prev){
         if (CHK_FLAG(actor->flags, ACTOR_FLAG_HIDDEN | ACTOR_FLAG_DISABLED)) {
            continue;
         }
-        
+
         if (CHK_FLAG(actor->flags, ACTOR_FLAG_PINNED)) {
             screen_x = SUBPX_TO_PX(actor->pos.x);
             screen_y = SUBPX_TO_PX(actor->pos.y);
         } else {
@@ -216,12 +217,12 @@
             continue;
         }
         SWITCH_ROM(actor->sprite.bank);
         spritesheet_t *sprite = actor->sprite.ptr;
-
+        UBYTE base_tile = actor->base_tile + (actor->using_sprite_buffer ? actor->reserve_tiles: 0);
         allocated_hardware_sprites += move_metasprite(
             *(sprite->metasprites + actor->frame),
-            actor->base_tile,
+            base_tile,
             allocated_hardware_sprites,
             screen_x,
             screen_y
         );
@@ -246,11 +247,11 @@
     if (!CHK_FLAG(actor->flags, ACTOR_FLAG_ACTIVE)) return;
     if (actor == &PLAYER) return;
     CLR_FLAG(actor->flags, ACTOR_FLAG_ACTIVE);
     DL_REMOVE_ITEM(actors_active_head, actor);
-	if (actors_active_tail == actor){
-		actors_active_tail = actor->prev;
-	}
+    if (actors_active_tail == actor){
+        actors_active_tail = actor->prev;
+    }
     DL_PUSH_HEAD(actors_inactive_head, actor);
     if ((actor->hscript_update & SCRIPT_TERMINATED) == 0) {
         script_terminate(actor->hscript_update);
     }
@@ -299,9 +300,9 @@
              if (CHK_FLAG(actor->flags, ACTOR_FLAG_PERSISTENT)) {
                 SET_FLAG(actor->flags, ACTOR_FLAG_DISABLED);
             } else {
                 return;
-            } 
+            }
         }
     }
 
     SET_FLAG(actor->flags, ACTOR_FLAG_ACTIVE);
@@ -549,5 +550,5 @@
     } else if (player_iframes != 0) {
         player_iframes--;
     }
     player_collision_actor = NULL;
-}
\ No newline at end of file
+}
